doc/decisions/error_message.md

```diff
@@ -0,0 +1,137 @@
+#Error message & handling concept
+
+## Problem
+
+The current error concept has disadvantages in following regards:
+
+- Too verbose error message
+- A lot of redundant errors
+- Hard to manage specification file
+- No senseful way for application developers to use error codes from elektra
+
+This concept should improve useability for end users and developers.
```

Comment:
Please also consider administrators, who will also read the error messages and are also capable of resolving resource errors and similar.

Reply:
Where exactly is the difference between an administrator and an end user? They both use elektra via gui/cmd/etc and fix a problem once it occurs. A developer reacts to error codes or uses the elektra API to emit an error.

Reply:
Of course there is overlap but I would not see an end user in the position to fix most resource error (e.g. resizing disks, changing permissions). So an application might add the text "please contact your system administrator" if an resource error occurs.

Having resource/conflict/temporary errors in different categories, however, is maybe the only requirement we need to add in order to consider system administrators.